packages/react-dom/src/__tests__/DOMPropertyOperations-test.js

```diff
@@ -322,28 +322,63 @@ describe('DOMPropertyOperations', () => {
     it('custom elements shouldnt have non-functions for on* attributes treated as event listeners', async () => {
       const container = document.createElement('div');
       const root = ReactDOMClient.createRoot(container);
+      // First render with nothing
+      await act(() => {
+        root.render(<my-custom-element />);
+      });
+      const customElement = container.querySelector('my-custom-element');
+      // Install a setter to activate the `in` heuristic
+      Object.defineProperty(customElement, 'onobj', {
+        set: function (x) {
+          this._onobj = x;
+        },
+        get: function () {
+          return this._onobj;
+        },
+      });
+      Object.defineProperty(customElement, 'onarray', {
+        set: function (x) {
+          this._onarray = x;
+        },
+        get: function () {
+          return this._onarray;
+        },
+      });
+      Object.defineProperty(customElement, 'ondate', {
+        set: function (x) {
+          this._ondate = x;
+        },
+        get: function () {
+          return this._ondate;
+        },
+      });
+      const onobj = {hello: 'world'};
+      const onarray = ['one', 'two'];
+      const ondate = new Date();
       await act(() => {
         root.render(
           <my-custom-element
             onstring={'hello'}
-            onobj={{hello: 'world'}}
-            onarray={['one', 'two']}
+            onobj={onobj}
+            onarray={onarray}
+            ondate={ondate}
```

Comment:
Should we also add a test case for setting null since you added logic for null?

Reply:
Good call. Added a couple more test cases.